### ISSUE TITLE
Failure when trying to store recarray with a non-packed (aligned) dtype

### ISSUE THREAD

**Issue by aldanor** [PyTables/PyTables#661]
If there are offsets in the dtype (which would often happen if the data came directly from C world), creating a table fails:

```python
>>> print(arr.dtype)
dtype({'names':['a','b','c'], 
       'formats':['<i8','i1','<i8'], 
       'offsets':[0,8,16], 'itemsize':24})
```

Here's what PyTables basically tries to do:

```python
try:
    ...
    wbufRA = numpy.rec.array(rows, dtype=self._v_dtype)
except Exception as exc:  # XXX
    raise ValueError("rows parameter cannot be converted into a "
                     "recarray object compliant with table '%s'. "
                     "The error was: <%s>" % (str(self), exc))
```

Here, `self._v_dtype` is this (which seems wrong):

```python
ipdb> p self._v_dtype
dtype([('a', '<i8'), ('b', 'i1'), ('c', '<i8')])
```

However, it's not the same layout, so `numpy.rec.array` ctor will fail:

```python
>>> np.rec.array(arr, dtype=np.dtype([('a', '<i8'), ('b', 'i1'), ('c', '<i8')]))
ValueError: new type not compatible with array.
```

Does that imply PyTables only supports packed structured dtypes or is it a bug in handling/parsing structured dtypes?

Thanks.

**Comment by FrancescAlted** [comment]
Hmm, yes, I would say that PyTables only deals with packed structured arrays (but have not looked into the code).  But in this case we would only need a way to copy the struct array with an 'offseted' dtype into one with no offsets (i.e. packed).  If you achieve that and want to provide a PR, we would be happy to accept it.

**Comment by FrancescAlted** [comment]
This should have been addressed in PR https://github.com/PyTables/PyTables/pull/720, and will be released in the forthcoming 3.5.0 PyTables release.
